ts/session/sending/PendingMessageCache.ts

```diff
@@ -1,36 +1,149 @@
+import { createOrUpdateItem, getItemById } from '../../../js/modules/data';
 import { RawMessage } from '../types/RawMessage';
 import { ContentMessage } from '../messages/outgoing';
+import { PubKey } from '../types';
+import * as MessageUtils from '../utils';
 
-// TODO: We should be able to import functions straight from the db here without going through the window object
+// This is an abstraction for storing pending messages.
+// Ideally we want to store pending messages in the database so that
+// on next launch we can re-send the pending messages, but we don't want
+// to constantly fetch pending messages from the database.
+// Thus we have an intermediary cache which will store pending messagesin
+// memory and sync its state with the database on modification (add or remove).
 
 export class PendingMessageCache {
-  private readonly cachedMessages: Array<RawMessage> = [];
+  public readonly isReady: Promise<boolean>;
+  private cache: Array<any>;
 
   constructor() {
-    // TODO: We should load pending messages from db here
+    // Load pending messages from the database
+    // You should await isReady on making a new PendingMessageCache
+    //   if you'd like to have instant access to the cache
+    this.cache = ['bleep'];
+
+    this.isReady = new Promise(async resolve => {
+      await this.loadFromDB();
+      resolve(true);
+    });
+  }
+
+  public getAllPending(): Array<RawMessage> {
+    // Get all pending from cache, sorted with oldest first
+    return [...this.cache].sort((a, b) => a.timestamp - b.timestamp);
   }
 
-  public addPendingMessage(
-    device: string,
+  public getForDevice(device: PubKey): Array<RawMessage> {
+    const pending = this.cache.filter(m => m.device === device.key);
```

Comment:
```suggestion
    return this.getAllPending().filter(m => m.device === device.key);
```
this will make this function a one liner as `getAllPending` returns sorted cache